ui/src/components/table/CustomTable.tsx

```diff
@@ -41,22 +58,34 @@ function CustomTable({
     sortDir,
     sortKey,
     tableConfig,
-}) {
-    const unifiedConfigs = getUnifiedConfigs();
-    const [entityModal, setEntityModal] = useState({ open: false });
-    const [deleteModal, setDeleteModal] = useState({ open: false });
+}) => {
+    const unifiedConfigs: GlobalConfig = getUnifiedConfigs();
+    const [entityModal, setEntityModal] = useState({
+        open: false,
+        serviceName: '',
```

Comment:
it looks ambiguous. serviceName is required for this state, but it is empty. Should it be optional instead?

Reply:
no, this is just for variable declaration.